server/src/policies.rs

```diff
@@ -199,75 +199,116 @@ impl Policies {
 impl VersionPolicy {
     pub fn from_yaml(config: &str) -> Result<Self> {
         let mut policies = Self::default();
-        let mut labels = vec![];
-
         let docs = YamlLoader::load_from_str(config)?;
         for config in docs.iter() {
-            for label in config["labels"].as_vec().get_or_insert(&[].into()).iter() {
-                let name = label["name"].as_str().ok_or_else(|| {
-                    anyhow::anyhow!("couldn't parse yaml: label without a name: {:?}", label)
-                })?;
-
-                labels.push(name.to_owned());
-                debug!("Applying policy for label {:?}", name);
-                let pattern = label["except_uri"].as_str().unwrap_or("^$"); // ^$ never matches; each path has at least a '/' in it.
-
-                match label["transform"].as_str() {
-                    Some("anonymize") => {
-                        policies.labels.insert(
+            match config {
+                Yaml::Hash(h) => {
+                    for (key, value) in h.iter() {
+                        let skey = key.as_str().unwrap_or("");
+                        match (skey, value) {
+                            ("labels", Yaml::Array(labels)) => policies.add_labels(labels)?,
+                            ("routes" | "endpoints", Yaml::Array(routes)) => {
+                                policies.add_routes(routes)?
+                            }
+                            ("labels" | "routes" | "endpoints", _) => {
+                                bail!("incorrect value for {skey}: {value:?}")
+                            }
+                            _ => bail!("unexpected yaml key: {key:?}"),
```

Comment:
just a nit, but I not a fan of unwrapping to an invalid value before matching on it. What do you think of that:
```suggestion
                        let skey = key.as_str();
                        match (skey, value) {
                            (Some("labels"), Yaml::Array(labels)) => policies.add_labels(labels)?,
                            (Some("routes" | "endpoints"), Yaml::Array(routes)) => {
                                policies.add_routes(routes)?
                            }
                            (Some("labels" | "routes" | "endpoints")), _) => {
                                bail!("incorrect value for {key}: {value:?}")
                            }
                            _ => bail!("unexpected yaml key: {key:?}"),
```